### PR TITLE
Pulsar Functions for Python docs

### PR DESCRIPTION
**Pull request by lucperkins**

### CODE REVIEW

**Comment by sijie** [comment]
@lucperkins what is the state of this PR? 

@srkukarni shall we include this for 2.0?

**Comment by srkukarni** [comment]
I think we need to complete this for 2.0

**Comment by lucperkins** [comment]
Work in progress. Will the Python API be complete by 2.0?

**Comment by lucperkins** [comment]
@srkukarni I know, I was waiting on counter support, but if that's not required for API completion then I can resume work on this.

**Comment by sijie** [comment]
we should leave counter/state part out of the scope of 2.0. let's finish the documentation for stateless part, we need finish the documentation for 2.0 by end of this week.

**Comment by lucperkins** [comment]
Ready for review

File: site/docs/latest/functions/api.md [hunk 1]
```diff
@@ -474,4 +495,205 @@ public class MetricRecorderFunction implements Function<Integer, Void> {
...
+
+#### Python native examples
+
+There is one example Python native function in [this folder](https://github.com/apache/incubator-pulsar/tree/master/pulsar-functions/python-examples):
+
+* [`pure_python_function_exclamation.py`](https://github.com/apache/incubator-pulsar/blob/master/pulsar-functions/python-examples/pure_python_function_exclamation.py)
+
+### Python SDK functions {#python-sdk}
+
+To get started developing Pulsar Functions using the Python SDK, you'll need to install
```
Comment (srkukarni) [inline site/docs/latest/functions/api.md]: broken sentence?

**Comment by lucperkins** [comment]
@srkukarni Fixed

**Pull request merged**